### DISCUSSION TITLE
Migration to Docker

### DISCUSSION THREAD

**Discussion by mcordes92** [eduardogsilva/wireguard_webadmin#30] [General]
We have currently set up a standard installation for "wireguard_webadmin" and would like to migrate to Docker. Could you assist us with the migration, as we use this system in our business operations?

**Comment by eduardogsilva** [comment]
Hello @mcordes92 

I tried to keep the information in the same place for easier backups and handling. If you didn't changed anything, you will find your sqlite db in /etc/wireguard/wireguard-webadmin-db.sqlite3  (make sure is exact the same name, the name changed on an earlier version due to issues with a migrate routine). If it's your case, you will need to rename to the new file, it should work anyway.

Actually, this is your most important file. Your firewall script and wg*.conf files are generated from the web interface with the information on this database. Make sure that you store this file in a safe place (and you can rollback to your old install if needed).

Go to your new machine, and follow the [Deployment instructions](https://github.com/eduardogsilva/wireguard_webadmin?tab=readme-ov-file#deployment-instructions). Make sure to use the nginx steps, as the alternative method use DEBUG mode and is not intended for production usage. 

After your new installation is working fine, make sure to STOP all related docker containers, and place your sqlite file inside the storage for your wireguard:/etc/wireguard volume. The default docker path for this volume is /var/lib/docker/volumes/wireguard_webadmin_wireguard/_data/ (it may change depending on your environment)


Cross your fingers and start wireguard_webadmin again. It "should" work. :D
Go to the web interface again, and your old admin password, should still be there. Go to the status page, click on the "update wireguard configuration" and then "restart wireguard service".

this will rebuild your wg*.conf files and the firewall script.

If you are using an older version, you may want to check the "new" firewall configuration page and migrate your old up/down scripts to this format ;)

Notes:

- After moving the db file to the new install, you may want to take a closer look on your first startup logs, as it will probably show your django migrations output.
- the default compose use the udp port range  "51820-51839". If you are using a different port range, you will need to adjust this.
- I don't actually expect any problems if you follow this steps, If the web interface starts fine with your "old" data, and you can't find any errors on the first startup during django migration, the rest should just work fine.
- You are sailing in uncharted waters, let me know how it goes.

Cheers and good luck!

**Comment by eduardogsilva** [comment]
@mcordes92 did you manage to migrate to docker?
